mistralrs-core/src/lib.rs

```diff
@@ -240,11 +277,81 @@ impl MistralRs {
             });
         });
 
-        this
+        Arc::new(Self {
+            sender,
+            log,
+            id,
+            creation_time: SystemTime::now()
+                .duration_since(UNIX_EPOCH)
+                .expect("Time travel has occurred!")
+                .as_secs(),
+            next_request_id: Mutex::new(RefCell::new(0)),
+            reboot_state,
+            engine_handler: RwLock::new(engine_handler),
+        })
+    }
+
+    /// attempts to reboot the engine, if the sender (only way to communicate with
+    /// the engine) is closed
+    fn reboot_engine(&self) -> Result<(), MistralRsError> {
+        let (new_sender, rx) = channel(10_000);
+        let reboot_state = self.reboot_state.clone();
+        let mut sender_lock = self.sender.write().map_err(|_| {
+            tracing::warn!("couldn't get write lock on the sender during reboot attempt");
```

Comment:
This is a small thing, but all the logging messages in mistralrs are capitalized :smile: If you could change them accordingly that would be great.